src/persist/src/operators/replay.rs

```diff
@@ -63,6 +70,16 @@ where
                     match snapshot {
                         Ok(snapshot) => {
                             let snapshot_since = snapshot.since();
+
+                            if PartialOrder::less_than(&as_of_frontier, &snapshot_since) {
```

Comment:
oh interesting, i was imagining that we'd have to thread this into the snapshot call. i don't see why this wouldn't work though